move contents of HACKING.md to a doc/HOWTO, and reference list of files

mcr opened the pull request:
##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
- [x] documentation is added or updated
- [ ] tests are added or updated

This change was a suggestion in #17926.

t8m commented on doc/HOWTO/adding-functions.md:
```diff
@@ -0,0 +1,36 @@
+
+ After that perform the following steps:
+
+    ./Configure -Werror --strict-warnings [your-options]
```
Please drop -Werror as strict warnings already adds that.

mcr replied:
removed in ea72ad8

levitte commented on doc/HOWTO/adding-functions.md:
```diff
@@ -0,0 +1,36 @@
+ either add a prototype in one of the existing OpenSSL header files;
+ or provide a new header file and edit
+ [Configurations/unix-Makefile.tmpl](Configurations/unix-Makefile.tmpl)
+ to pick up that file.
```
The part about `Configurations/unix-Makefile.tmpl` is *aged*, apart for some very special cases, which should be dealt with on a case by case basis and not worth mentioning here.  So the part "and edit . ...  to pick up that file." can simply be removed.

mcr replied:
removed in f370ab3

levitte commented:
I just noticed that this file isn't referred to from anywhere...  is there a risk that it will get lost in oblivion?

mcr commented:
> I just noticed that this file isn't referred to from anywhere... is there a risk that it will get lost in oblivion?

Was HACKING.md referenced before?
It's at the top-level, so maybe it would get noticed, and it points at the other files.

levitte commented:
> > I just noticed that this file isn't referred to from anywhere... is there a risk that it will get lost in oblivion?
> 
> Was HACKING.md referenced before? It's at the top-level, so maybe it would get noticed, and it points at the other files.

I hasn't been referenced before, we did indeed rely on its top-level location for it to be noticed.  This move makes it less directly visible...

mcr commented:
> > > I just noticed that this file isn't referred to from anywhere... is there a risk that it will get lost in oblivion?
> > 
> > 
> > Was HACKING.md referenced before? It's at the top-level, so maybe it would get noticed, and it points at the other files.
> 
> I hasn't been referenced before, we did indeed rely on its top-level location for it to be noticed. This move makes it less directly visible...

The file is STILL THERE.  It just references content elsewhere now.

mcr committed:
move contents of HACKING.md to a doc/HOWTO, and reference list of files
removed -Werror from instructions on adding functions
rename HACKING.md to HOWTO.md

fwh-dc reviewed (changes requested):
Could we call the top-level file HOWTO.md? Hacking is not the correct terminology imo.

mcr commented on HOWTO.md:
```diff
@@ -0,0 +1,10 @@
+* [doc/HOWTO/adding-functions.md](Adding new Functions)
+* [doc/HOWTO/documenting-functions-macros.md](Documenting Functions and Macros)
+
+>>>>>>> d3e6afd41a (move contents of HACKING.md to a doc/HOWTO, and reference list of files)
```
I hate it when that happens.

fwh-dc reviewed (dismissed):
Approved with one trivial nit pending. Ready for second review.

fwh-dc reviewed (approved):
LGTM

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 30 days ago

tom-cosgrove-arm commented on doc/HOWTO/adding-functions.md:
```diff
@@ -1,5 +1,5 @@
+ADDING FUNCTIONS to OPENSSL
+===========================
 
 This document describes the way to add custom modifications to OpenSSL
```
"custom modifications" sounds like local changes, as opposed to adding functions to the main repo. Should it be something like?

```
This document describes how to add new functions to the OpenSSL sources.
```

openssl-machine commented:
This pull request is ready to merge

npajkovsky commented:
Thank you. Merged into `master`.
